test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/ProjectSystems/LegacyPackageReferenceProjectTests.cs

```diff
@@ -1129,6 +1129,170 @@ public async Task GetInstalledVersion_WithoutPackages_WithAssets_ReturnsEmpty()
             }
         }
 
+        [Fact]
+        public async Task GetTransitivePackagesAsync_WithTransitivePackageReferences_ReturnsPackageIdentities()
+        {
+            using (var testDirectory = TestDirectory.Create())
+            {
+                // Setup
+                LegacyPackageReferenceProject testProject = CreateLegacyPackageReferenceProject(testDirectory, "[1.0.0, )");
+
+                var settings = NullSettings.Instance;
+                var context = new DependencyGraphCacheContext(NullLogger.Instance, settings);
+
+                var packageSpecs = await testProject.GetPackageSpecsAsync(context);
+
+                // Package directories
+                var sources = new List<PackageSource>();
+                var packagesDir = new DirectoryInfo(Path.Combine(testDirectory, "globalPackages"));
+                var packageSource = new DirectoryInfo(Path.Combine(testDirectory, "packageSource"));
+                packagesDir.Create();
+                packageSource.Create();
+                sources.Add(new PackageSource(packageSource.FullName));
+
+                var logger = new TestLogger();
+                var request = new TestRestoreRequest(packageSpecs[0], sources, packagesDir.FullName, logger)
+                {
+                    LockFilePath = Path.Combine(testDirectory, "obj", "project.assets.json")
+                };
+
+                await SimpleTestPackageUtility.CreateFullPackageAsync(packageSource.FullName, "packageB", "1.0.0");
+                await SimpleTestPackageUtility.CreateFullPackageAsync(
+                    packageSource.FullName,
+                    "packageA",
+                    "2.15.3",
+                    new Packaging.Core.PackageDependency[]
+                    {
+                        new Packaging.Core.PackageDependency("packageB", VersionRange.Parse("1.0.0"))
+                    });
+
+                // Act
+                var command = new RestoreCommand(request);
+                var result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
+                Assert.True(result.Success);
+                var packages = await testProject.GetAllPackagesAsync(CancellationToken.None);
+                var cache_packages = await testProject.GetAllPackagesAsync(CancellationToken.None);
```

Comment:
This test doesn't do anything to validate that the packages were actually cached. So if there was a bug and the assets file is re-read every time, this test wouldn't fail. Hence, calling `GetAllPackagesAsync` and asseting the contents doesn't actually prevent any regressions.

Reply:
Added a check that the assets file was not accessed when we were expecting the packages to come from the cache.